doc/decisions/header_file_structure.md

```diff
@@ -12,31 +12,135 @@ The big problem is kdbprivate.h. It has two main problems:
 
 ## Constraints
 
+- Each library should have at least one separate header file:
+
+  Without this constraint, we would constantly include things that aren't needed.
+  This is inconvenient as it clutters the auto-suggestions of IDEs.
+  It also slows down the compile process, since the compiler has to parse everything that was included (see [also](https://lore.kernel.org/lkml/YdIfz+LMewetSaEB@gmail.com/T/)).
```

Comment:
The link is misleading here. This was a huge undertaking and has little to do what is planned with this decision. Please move it down to Notes.